packages/changelog-github/src/index.ts

```diff
@@ -46,23 +46,73 @@ const changelogFunctions: ChangelogFunctions = {
         'Please provide a repo to this changelog generator like this:\n"changelog": ["@changesets/changelog-github", { "repo": "org/repo" }]'
       );
     }
-    const [firstLine, ...futureLines] = changeset.summary
+
+    let prFromSummary: number | undefined;
+    let commitFromSummary: string | undefined;
+    let userFromSummary: string | undefined;
+
+    const replacedChangelog = changeset.summary
+      .replace(/\n\s*(?:pr|pull|pull\s+request):\s*#?(\d+)/i, (_, pr) => {
```

Comment:
note that instead of matching `\n` you could match `^` with the `m` flag since then `^` means the start of a line instead of the start of the string
```suggestion
      .replace(/^\s*(?:pr|pull|pull\s+request):\s*#?(\d+)/mi, (_, pr) => {
```